_posts/2020-09-01-weather-prediction/weather_prediction_deep_learning.Rmd

```diff
@@ -0,0 +1,695 @@
+
+Directions on how to improve on that baseline are given by recent publications. Weyn et al.[@weyn], in addition to applying
+more geometrically-adequate spatial preprocessing, use a U-Net-based architecture instead of a plain convnet. Rasp and Thuerey
+[@rasp2020purely], building on a fully convolutional, high-capacity ResNet architecture, add a key new procedural ingredient:
+pre-training on climate models. With their method, they are able to not just compete with physical models, but also, show
+evidence of the network learning about physical structure and dependencies. Unfortunately, compute facilities of this order
+are not available to the average individual, which is why we'll content ourselves with demonstrating a simple toy model.
+Still, having seen a simple model in action, as well as the type of data it works on, should help a lot in understanding how
+DL can be used for weather prediction.
+
+## Dataset
+
+[Weatherbench](https://github.com/pangeo-data/WeatherBench) was explicitly created as a benchmark dataset and thus, as is
+common for this species, hides a lot of preprocessing and standardization effort from the user. Atmospheric data are available
+on an hourly basis, ranging from 1979 to 2018, at different spatial resolutions. Depending on resolution, there are about 15
+to 20 measured variables, including temperature, geopotential, wind speed, and humidity. Of these variables, some are
+available at several pressure levels. Thus, our example makes use of a small subset of available "channels". To save storage,
+network and computational resources, it also operates at the smallest available resolution.
+
+This post is accompanied by executable code on [Google
+Colaboratory](https://colab.research.google.com/drive/1URSjWfcnhgesHWCqd7Un5GHECYblY04r?usp=sharing), which should not just
+render unnecessary any copy-pasting of code snippets but also, allow for uncomplicated modification and experimentation.
+
+To read in and extract the data, stored as [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) files, we use
+[tidync](https://docs.ropensci.org/tidync/), a high-level package built on top of
+[ncdf4](https://cran.r-project.org/package=ncdf4) and [RNetCDF](https://cran.r-project.org/package=RNetCDF). Otherwise,
+availability of the usual "TensorFlow family" as well as a subset of tidyverse packages is assumed.
+
+```{r}
+library(reticulate)
+library(tensorflow)
+library(keras)
+library(tfdatasets)
+library(tfautograph)
+
+library(dplyr)
+library(ggplot2)
+library(tidyr)
+library(purrr)
+library(lubridate)
+
+library(tidync)
+```
+
+As already alluded to, our example makes use of two spatio-temporal series: 500hPa geopotential and 850hPa temperature. The
+following commands will download and unpack the respective sets of by-year files, for a spatial resolution of 5.625 degrees:
+
+```{r}
+system('wget "https://dataserv.ub.tum.de/s/m1524895/download?path=%2F5.625deg%2Fgeopotential_500&files=geopotential_500_5.625deg.zip" \\
```

Comment:
would you consider using `download.file` and `dir.create` etc. This doesn't run on Windows...

Reply:
oh, right :-) it's fine on Colab but not everywhere else ;-)